internal/job/executor.go

```diff
@@ -134,7 +133,7 @@ func (e *Executor) Run(ctx context.Context) (exitCode int) {
 
 	// Tear down the environment (and fire pre-exit hook) before we exit
 	defer func() {
-		if err = e.tearDown(ctx); err != nil {
+		if err = e.tearDown(cancelCtx); err != nil {
```

Comment:
this is the exception to my comment above - `tearDown` still happens after a job is cancelled, and so shouldn't be called with a cancellable context.

i think we'll need to rename the context before the call to `context.WithCancel` to something like `nonCancelContext`, and then pass the non-cancel context into `tearDown`